hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/HardLink.java

```diff
@@ -204,6 +209,20 @@ public static void createHardLinkMult(File parentDir, String[] fileBaseNames,
     }
   }
 
+    /**
+     * Determines whether the system supports hardlinks.
+     * @param f - file to examine
+     * @return true if hardlinks are supported, false otherwise
+     */
+  public static boolean supportsHardLink(File f) {
+    try {
+      FileStore store = Files.getFileStore(f.toPath());
+      return store.supportsFileAttributeView(FileAttributeView);
+    } catch (IOException e) {
+      return false;
```

Comment:
We should add some logs.

Reply:
I have added a Warn Log when we catch the exception.